### PR TITLE
add --target-tag as argument for dca hacky image build

### PR DESCRIPTION
**Pull request by adel121**
### What does this PR do?

Add `--target-tag` as an argument for `cluster-agent.hacky-dev-image-build` invoke command.

### Motivation

Be consistent with `agent.hacky-dev-image-build` invoke command. 

### Describe how to test/QA your changes

Veriication:

```
inv cluster-agent.hacky-dev-image-build --target-image=cluster-agent --target-tag=custom-tag        
--- Setting rtloader paths to lib:/workspaces/datadog-agent/dev/lib | header:/workspaces/datadog-agent/dev/include | common headers:
Successfully wrote /workspaces/datadog-agent/Dockerfiles/cluster-agent/datadog-cluster.yaml
Loading release versions for nightly-a7
Security Agent polices for nightly-a7: master
Cloning into 'Dockerfiles/cluster-agent/security-agent-policies'...
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            Install the buildx component to build images with BuildKit:
            https://docs.docker.com/go/buildx/

Sending build context to Docker daemon  1.328GB
Step 1/16 : FROM ubuntu:latest AS src
 ---> a50ab9f16797
Step 2/16 : COPY . /usr/src/datadog-agent
 ---> 28ea334ec26d
Step 3/16 : RUN find /usr/src/datadog-agent -type f \! -name \*.go -print0 | xargs -0 rm
 ---> Running in 11ebb2d373d6
 ---> Removed intermediate container 11ebb2d373d6
 ---> 4505b347626a
Step 4/16 : RUN find /usr/src/datadog-agent -type d -empty -print0 | xargs -0 rmdir
 ---> Running in cb7b43e50ee8
 ---> Removed intermediate container cb7b43e50ee8
 ---> cf113d458154
Step 5/16 : FROM golang:latest AS dlv
 ---> 9cbeef2f2690
Step 6/16 : RUN go install github.com/go-delve/delve/cmd/dlv@latest
 ---> Using cache
 ---> f82655ff2d87
Step 7/16 : FROM gcr.io/datadoghq/cluster-agent:7.51.1
 ---> 0b3a6698d5a6
Step 8/16 : ENV DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> b977e2465dea
Step 9/16 : RUN apt-get update &&     apt-get install -y bash-completion less vim tshark &&     apt-get clean
 ---> Using cache
 ---> b17b6e5afa1e
Step 10/16 : ENV DELVE_PAGER=less
 ---> Using cache
 ---> 18f2de3a8b47
Step 11/16 : COPY --from=dlv /go/bin/dlv /usr/local/bin/dlv
 ---> Using cache
 ---> 5a68c30ab086
Step 12/16 : COPY --from=src /usr/src/datadog-agent /workspaces/datadog-agent
 ---> Using cache
 ---> 29a7b4b1f7b2
Step 13/16 : COPY bin/datadog-cluster-agent/datadog-cluster-agent /opt/datadog-agent/bin/datadog-cluster-agent
 ---> db737b8c0acc
Step 14/16 : RUN agent                 completion bash > /usr/share/bash-completion/completions/agent
 ---> Running in f3c17b783842
 ---> Removed intermediate container f3c17b783842
 ---> 8ef0db0b2c42
Step 15/16 : RUN datadog-cluster-agent completion bash > /usr/share/bash-completion/completions/datadog-cluster-agent
 ---> Running in 7d3664c3e33f
 ---> Removed intermediate container 7d3664c3e33f
 ---> 6dbba173a81d
Step 16/16 : ENV DD_SSLKEYLOGFILE=/tmp/sslkeylog.txt
 ---> Running in 2bcd68558907
 ---> Removed intermediate container 2bcd68558907
 ---> 51ee3535ec21
Successfully built 51ee3535ec21
Successfully tagged cluster-agent:custom-tag
```

### CODE REVIEW

**Review by clamoriniere — APPROVED** [review]

**Comment by pducolin** [comment]
Hi @adel121 , the `qa/skip-qa` label is deprecated, if there were no code changes can you please add `qa/no-code-changes`? This help us triage why PR can skip the QA

**Review by clamoriniere — APPROVED** [review]

File: tasks/cluster_agent.py [hunk 1]
```diff
@@ -210,11 +217,11 @@ def hacky_dev_image_build(ctx, base_image=None, target_image="cluster-agent", pu
...
-
+        target_image_name = f'{target_image}:{target_tag}'
         pull_env = {}
         if signed_pull:
             pull_env['DOCKER_CONTENT_TRUST'] = '1'
-        ctx.run(f'docker build -t {target_image} -f {dockerfile.name} .', env=pull_env)
+        ctx.run(f'docker build -t {target_image_name} -f {dockerfile.name} .', env=pull_env)
 
         if push:
             ctx.run(f'docker push {target_image}')
```
Comment (chouetz) [inline tasks/cluster_agent.py:227]:
```suggestion
            ctx.run(f'docker push {target_image_name}')
```
I suppose we want to push with the tagged name, correct?

Comment (adel121) [reply tasks/cluster_agent.py:227]: Thanks for catching this

**Review by chouetz — APPROVED** [review]

**Review by KevinFairise2 — APPROVED** [review]

**Pull request merged**